### PR TITLE
Revert support object and class string as `$rules` in `Validator::validate()`

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ✔️
| Fixed issues  | -

### CODE REVIEW

File: src/ValidatorInterface.php [hunk 1]
```diff
@@ -7,7 +7,7 @@
 /**
  * Validator validates {@link DataSetInterface} against rules set for data set attributes.
  *
- * @psalm-type RulesType = null|RuleInterface|RulesProviderInterface|callable|iterable<RuleInterface|RuleInterface[]|callable|callable[]>
```
Comment (arogachev) [inline src/ValidatorInterface.php:10]: These types can be mentioned in PHPDoc now. But I can handle it later during adjusting docs.

**Review by arogachev — CHANGES_REQUESTED** [review]
:+1:

**Comment by arogachev** [comment]
Related - #400.

File: src/ValidatorInterface.php (hunk 1 above)
Comment (vjik) [reply src/ValidatorInterface.php:10]: Yes. Here need list types with descriptions.

File: src/Helper/RulesNormalizer.php [hunk 2]
```diff
@@ -82,12 +85,11 @@ private static function prepareRulesArray(
             return [$rules];
         }
 
-        /** @psalm-suppress RedundantConditionGivenDocblockType */
         if (is_iterable($rules)) {
             return $rules;
         }
 
-        throw new InvalidArgumentException('A rules object must implement RulesProviderInterface or RuleInterface.');
+        return (new AttributesRulesProvider($rules))->getRules();
```
[the remark this answers is not in the document]
Comment (arogachev) [reply src/Helper/RulesNormalizer.php:92]: Exactly. Then it's OK.

**Review by samdark — APPROVED** [review]

**Pull request merged**